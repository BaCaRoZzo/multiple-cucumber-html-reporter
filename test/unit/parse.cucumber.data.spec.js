const {parseFeatureHooks, parseMetadata} = require('../../lib/parse.cucumber.data');

describe('parseFeatureHooks', () => {
    it('should be able add a before hook', () => {
        expect(parseFeatureHooks(
            [
                {
                    result: {
                        status: 'passed'
                    },
                    match: {}
                }
            ],
            'before',
        ))
            .toMatchSnapshot();
    });

    it('should be able add an after hook', () => {
        expect(parseFeatureHooks(
            [

                {
                    arguments: [
                        {
                            foo: 'bar'
                        }
                    ],
                    embeddings: [
                        {
                            bar: 'foo'
                        }
                    ],
                    result: {
                        'foo-bar': true,
                    },
                    match: {
                        location: 'something'
                    }
                }
            ],
            'after',
        ))
            .toMatchSnapshot();
    });

    it('should be able add multiple before hooks', () => {
        expect(parseFeatureHooks(
            [
                {
                    result: {
                        status: 'passed'
                    },
                    match: {}
                },
                {
                    result: {
                        status: 'failed'
                    },
                    match: {
                        location: 'something'
                    }
                }
            ],
            'before',
        ))
            .toMatchSnapshot();
    });
});

describe('parseMetadata', () => {
    it('should be able add the default metadata', () => {
        expect(parseMetadata(
            {
                foo: 'bar'
            }
        ))
            .toMatchSnapshot();
    });

    it('should be able add the provided metadata', () => {
        expect(parseMetadata(
            {
                foo: 'bar'
            },
            {
                metadata: {
                    foo: 'bar'
                }
            }
        ))
            .toMatchSnapshot();
    });

    it('should not add metadata if it is already there', () => {
        expect(parseMetadata(
            {
                foo: 'bar',
                metadata: {
                    fooBar: 'fooBar'
                }
            },
            {
                metadata: {
                    foo: 'bar'
                }
            }
        ))
            .toMatchSnapshot();
    });
});
