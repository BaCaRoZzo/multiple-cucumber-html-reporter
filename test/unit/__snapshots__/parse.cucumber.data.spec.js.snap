// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseFeatureHooks should be able add a before hook 1`] = `
Array [
  Object {
    "arguments": Array [],
    "embeddings": Array [],
    "keyword": "before",
    "line": "",
    "match": Object {
      "location": "can not be determined",
    },
    "name": "Hook",
    "result": Object {
      "status": "passed",
    },
  },
]
`;

exports[`parseFeatureHooks should be able add an after hook 1`] = `
Array [
  Object {
    "arguments": Array [
      Object {
        "foo": "bar",
      },
    ],
    "embeddings": Array [
      Object {
        "bar": "foo",
      },
    ],
    "keyword": "after",
    "line": "",
    "match": Object {
      "location": "something",
    },
    "name": "Hook",
    "result": Object {
      "foo-bar": true,
    },
  },
]
`;

exports[`parseFeatureHooks should be able add multiple before hooks 1`] = `
Array [
  Object {
    "arguments": Array [],
    "embeddings": Array [],
    "keyword": "before",
    "line": "",
    "match": Object {
      "location": "can not be determined",
    },
    "name": "Hook",
    "result": Object {
      "status": "passed",
    },
  },
  Object {
    "arguments": Array [],
    "embeddings": Array [],
    "keyword": "before",
    "line": "",
    "match": Object {
      "location": "something",
    },
    "name": "Hook",
    "result": Object {
      "status": "failed",
    },
  },
]
`;

exports[`parseMetadata should be able add the default metadata 1`] = `
Object {
  "foo": "bar",
  "metadata": Object {
    "browser": Object {
      "name": "not known",
      "version": "not known",
    },
    "device": "not known",
    "platform": Object {
      "name": "not known",
      "version": "not known",
    },
  },
}
`;

exports[`parseMetadata should be able add the provided metadata 1`] = `
Object {
  "foo": "bar",
  "metadata": Object {
    "metadata": Object {
      "foo": "bar",
    },
  },
}
`;

exports[`parseMetadata should not add metadata if it is already there 1`] = `
Object {
  "foo": "bar",
  "metadata": Object {
    "fooBar": "fooBar",
  },
}
`;
